Automattic/jetpack
# Sharing: switch to only a base64-encoded icon font

## Proposed changes:

Follow-up to #28849

In #28849, we moved away from using a base64-encoded font and an eot file for multiple reasons:

- https://github.com/Automattic/jetpack/issues/6076
- https://github.com/Automattic/jetpack/issues/7148
- https://github.com/Automattic/jetpack/pull/28694#issuecomment-1423000642

Instead, we switched to loading 3 files from local paths:

- `woff2`, supported by all modern browsers: https://caniuse.com/woff2
- `woff` and `ttf`, around for compatibility with old browsers: https://css-tricks.com/snippets/css/using-font-face-in-css/

Given how widespread `woff2` support looks like today, we may be able to use just that. We will, however, need to revert to a base64-encoded font to avoid CORS issues when the files are served from a CDN, like Photon CDN:

```
Access to font at 'https://c0.wp.com/p/jetpack/11.9/_inc/social-logos/social-logos.woff2' from origin 'https://yoursite.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

Both approaches seem to be problematic:

- base64-encoded fonts are broken for some people.
- fonts loaded via a path have CORS issues for everyone using a CDN. => **This seems like the biggest problem for us right now, given that Jetpack includes such a CDN.**

This commit is consequently not a perfect solution. Switching to SVGs, as discussed in #28861, seems to be the path forward.

> **Note**
> This PR edits a file that was previously auto-generated in https://github.com/Automattic/social-logos . I wouldn't normally want to manually edit auto-generated files, but the social-logos build tools are not working anymore (see https://github.com/Automattic/social-logos/issues/116). As a result, I think it would be fine to apply this temporary fix here until we get rid of the icon font entirely.

### Other information:


## Jetpack product discussion

* p1678548282092199-slack-CBG1CP4EN

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

**Test with different browsers**

* Start with a site connected to WordPress.com, **and running the 11.9 stable release of Jetpack**.
* Go to Jetpack > Settings > Performance, and enable the site accelerator, specifically the static file CDN (second option).
* Go to Jetpack > Settings > Sharing and enable Sharing buttons
* Configure sharing buttons: add a Mastodon button to the list.
* View a post on the frontend; the Mastodon icon should be broken.
* You should also see the CORS error mentioned above in your browser console.
* Open your browser dev tools, and in the sources tab, find the `social-logos.min.css` file under c0.wp.com > p > jetpack > 11.9 > _inc > social-logos
* Edit that file, paste in the modified version from this PR.
* The Mastodon icon should now look good.
